### ISSUE TITLE
exchange_name arg not effective

### ISSUE THREAD

**Issue by cbogithub** [enigmampc/catalyst#165]
Dear Catalyst Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment

* Operating System: (Windows Version or `$ uname --all`)
window 10
* Python Version: `$ python --version`
python3.6

* Python Bitness: `$ python -c 'import math, sys;print(int(math.log(sys.maxsize + 1, 2) + 1))'`
x64
* How did you install Catalyst: (`pip`, `conda`, or `other (please explain)`)
* Python packages: `$ pip freeze` or `$ conda list`
aiodns==1.1.1
aiohttp==2.3.7
alabaster==0.7.7
alembic==0.9.6
aniso8601==1.0.0
asciichart==0.1
asciichartpy==1.5.7
asn1crypto==0.24.0
astroid==1.5.3
async-timeout==2.0.0
attrs==17.4.0
autobahn==17.10.1
automat==0.6.0
Babel==2.2.0
banal==0.3.3
bcolz==0.12.1
beautifulsoup4==4.6.0
bleach==1.5.0
blinker==1.4
bokeh==0.12.13
boto3==1.5.14
botocore==1.8.28
Bottleneck==1.2.1
browsermob-proxy==0.8.0
cachepy==0.1
cchardet==2.1.1
ccxt==1.10.620
certifi==2017.11.5
cffi==1.11.2
chardet==3.0.4
click==6.6
colorama==0.3.7
constantly==15.1.0
contextlib2==0.5.5
coverage==4.0.3
cryptography==2.1.4
cx-freeze==5.0
cycler==0.10.0
cymem==1.31.2
cyordereddict==1.0.0
Cython==0.27.3
cytoolz==0.8.2
dash==0.19.0
dash-core-components==0.17.0
dash-html-components==0.8.0
dash-renderer==0.11.1
dataset==1.0.5
decorator==4.1.2
defusedxml==0.4.1
dill==0.2.7.1
Django==1.10.2
django-autocomplete-light==3.2.0
django-extensions==1.7.4
django-gm2m==0.4.7
django-querysetsequence==0.7
django-tagging==0.4.5
django-taggit==0.21.3
docutils==0.12
dropbox==6.4.0
ecdsa==0.13
empyrical==0.2.2
enum34==1.1.6
facebook-sdk==2.0.0
fbchat==1.0.19
Flask==0.12.2
Flask-API==0.6.6.post1
Flask-AppBuilder==1.7.0
Flask-Auth==0.85
Flask-BabelPkg==0.9.6
Flask-BasicAuth==0.2.0
Flask-Bootstrap3==3.1.1.3
Flask-Compress==1.4.0
Flask-Cors==2.1.0
Flask-Dropbox==0.3
Flask-Excel==0.0.4
Flask-GoogleAuth==0.4.2
Flask-GoogleMaps==0.1.9
Flask-Headers==1.0
Flask-HTTPAuth==3.1.2
Flask-HTTPS==0.1.0
Flask-JSON==0.3.0
Flask-Jsonpify==1.5.0
Flask-JWT==0.3.2
Flask-Login==0.2.11
Flask-Mail==0.9.1
Flask-Navigation==0.2.0
Flask-OpenID==1.2.5
Flask-Principal==0.4.0
Flask-RESTful==0.3.4
flask-secure-cookie==0.1.2
flask-secure-headers==0.6
Flask-Security==1.7.5
Flask-ServiceLayer==0.0.4
Flask-Session==0.2.3
Flask-SocketIO==2.7.1
Flask-SQLAlchemy==2.0
Flask-WTF==0.12
ftfy==4.4.3
future==0.15.2
geopy==1.11.0
gevent==1.1.2
google-api-python-client==1.6.3
gpsoauth==0.4.0
greenlet==0.4.5
html5lib==0.9999999
httplib2==0.10.3
hyperlink==17.3.1
idna==2.6
imutils==0.4.3
incremental==17.5.0
intervaltree==2.1.0
IPy==0.83
ipython-genutils==0.2.0
isort==4.2.15
itsdangerous==0.24
Jinja2==2.8
jmespath==0.9.3
joblib==0.11
jsonschema==2.6.0
jupyter-core==4.4.0
Keras==2.0.8
Kivy-Garden==0.1.4
lazy-object-proxy==1.3.1
logbook==1.1.0
lru-dict==1.1.6
lxml==4.1.1
Mako==1.0.7
Markdown==2.6.9
MarkupSafe==0.23
matplotlib==2.1.0
mccabe==0.6.1
MechanicalSoup==0.9.0.post4
mock==2.0.0
multidict==3.3.2
multipledispatch==0.4.9
murmurhash==0.26.4
nbformat==4.4.0
networkx==2.0
nltk==3.2.5
normality==0.5.7
nose==1.3.7
Nuitka64==5.1.190
numexpr==2.6.4
numpy==1.13.3+mkl
oauth2client==4.1.2
olefile==0.44
opencv-contrib-python==3.3.0.9
opencv-python==3.3.0.9
pandas==0.18.1
pandas-datareader==0.5.0
paramiko==1.16.0
passlib==1.6.5
pathlib==1.0.1
patsy==0.5.0
pbr==1.10.0
peewee==2.8.1
picamera==1.12
Pillow==4.3.0
plac==0.9.6
plotly==2.2.3
pluggy==0.3.1
preshed==1.0.0
protobuf==3.4.0
protobuf-to-dict==0.1.0
py==1.4.34
py2exe==0.9.2.0
pyasn1==0.4.2
pyasn1-modules==0.2.1
pycares==2.3.0
pycparser==2.18
pycryptodome==3.4.6
pycurl==7.19.3.1
pyexcel==0.2.2
pyexcel-io==0.2.0
pyexcel-webio==0.0.7
Pygments==2.1.1
PyJWT==1.4.0
PyLD==0.7.1
pylint==1.7.4
pymessenger==0.0.7.0
pymongo==3.5.1
PyOpenGL==3.1.0
PyOpenGL-accelerate==3.1.0
pyOpenSSL==17.5.0
pyparsing==2.2.0
pypiwin32==220
pypng==0.0.18
pyqrcode==1.2.1
PyQt5==5.9
pyreadline==2.1
pytest==3.2.2
pytest-cov==2.4.0
pytest-django==3.0.0
python-binance==0.5.17
python-crfsuite==0.9.5
python-dateutil==2.6.1
python-editor==1.0.3
python-engineio==1.0.3
python-socketio==1.6.0
python-telegram-bot==8.1.1
python-wordpress-xmlrpc==2.3
python3-openid==3.0.10
pytz==2017.3
PyYAML==3.12
redis==2.10.6
redo==1.6
regex==2017.9.23
requests==2.18.4
requests-file==1.4.3
requests-ftp==0.3.1
requests-toolbelt==0.8.0
robobrowser==0.5.3
rsa==3.4.2
s2sphere==0.2.4
s3transfer==0.1.12
scapy==2.3.2
scikit-learn==0.19.0
scipy==1.0.0rc1
selenium==3.8.0
service-identity==17.0.0
sip==4.19.3
six==1.11.0
sklearn==0.0
snowballstemmer==1.2.1
sortedcontainers==1.5.9
spacy==1.9.0
speaklater==1.3
Sphinx==1.3.5
sphinx-rtd-theme==0.1.9
splinter==0.7.4
SQLAlchemy==1.2.0
statsmodels==0.8.0
TA-Lib==0.4.15
tables==3.4.2
tensorflow==1.3.0
tensorflow-tensorboard==0.1.8
termcolor==1.1.0
texttable==0.8.4
thinc==6.5.2
toolz==0.8.2
tornado==4.5.3
tox==2.3.1
tqdm==4.19.1.post1
traitlets==4.3.2
twisted==17.9.0
txaio==2.8.2
ujson==1.35
update==0.4.4
uritemplate==3.0.0
urllib3==1.22
virtualenv==14.0.6
wcwidth==0.1.7
websocket==0.2.1
websocket-client==0.44.0
Werkzeug==0.12.2
winrandom-ctypes==1.0.3
wrapt==1.10.11
WTForms==2.1
xxhash==0.6.1
yarl==0.17.0
zipline==1.1.1
zope.interface==4.4.3
Now that you know a little about me, let me tell you about the issue I am
having:

# Description of Issue

I changed exchange_name arg in run_algorithm(..) to " binance" catalyst still run with exchange_name arg is "Poloniex" and even though i changed exchange_name  to anything  catalyst  still run with "Poloniex".
so, what problem in here ?

my script:
```

from catalyst.api import symbol
from catalyst.utils.run_algo import run_algorithm


def initialize(context):
    context.asset = symbol('bcc_usdt')


def handle_data(context, data):
    data.history(context.asset, ['close'], bar_count=100, frequency='5T')


def analyze(context=None, results=None):
    pass


if __name__ == '__main__':
    run_algorithm(
        capital_base=100,
        initialize=initialize,
        handle_data=handle_data,
        analyze=analyze,
        exchange_name='binance',
        algo_namespace="binance_is_broken",
        base_currency='usdt',
        data_frequency='minute',
        simulate_orders=True,
live=True)
```
erorr log: https://pastebin.com/pf7xjptv

* What did you expect to happen?
* What happened instead?

Here is how you can reproduce this issue on your machine:

## Reproduction Steps

1.
2.
3.
...

## What steps have you taken to resolve this already?

...

# Anything else?

...

Sincerely,
`$ whoami`

**Comment by fredfortier** [comment]
It’s probably using poloniex to fetch the benchmark data, not to run your
algorithm.

**Comment by cbogithub** [comment]
how to fix it ?

**Comment by fredfortier** [comment]
It’s the expected behavior (at least for now). The benchmark is the value
of Bitcoin. Is this an issue because of connectivity issues with Poloniex?

**Comment by cbogithub** [comment]
temp fix :
 change "poloniex" in line 147 in file "site-packages\catalyst\data\loader.py" to "binance" or other exchange 

and it working now

**Comment by lacabra** [comment]
This is the expected behavior as @fredfortier mentions above, as the algorithm will trade agains the exchange that you specify in the `exchange_name` parameter, but uses Poloniex to fetch a benchmark to compute the performance of your algorithm. Thus, this is a duplicate of #163. Closing this issue in favor of the previous one. This issue has been fixed in commit cf6c3bb76b336e3d3c16c7b47bc3ef164f7f9ef3 and will be included the next upcoming release.
